cli/internal/cmd/configgenerate.go

```diff
@@ -155,3 +151,19 @@ func generateCompletion(cmd *cobra.Command, args []string, toComplete string) ([
 		return []string{}, cobra.ShellCompDirectiveError
 	}
 }
+
+// resolveK8sVersion takes the user input from --kubernetes and transforms a MAJOR.MINOR definition into a supported
+// MAJOR.MINOR.PATCH release.
+func resolveK8sVersion(k8sVersion string) (string, error) {
+	prefixedVersion := compatibility.EnsurePrefixV(k8sVersion)
+	if !semver.IsValid(prefixedVersion) {
+		return "", fmt.Errorf("kubernetes flag does not specify a valid semantic version: %s", k8sVersion)
+	}
+
+	extendedVersion := config.K8sVersionFromMajorMinor(prefixedVersion)
+	if extendedVersion == "" {
+		return "", fmt.Errorf("--kubernetes (%s) does not specify a valid Kubernetes version. Supported versions: %s", strings.TrimPrefix(k8sVersion, "v"), supportedVersions())
```

Comment:
```suggestion
		return "", fmt.Errorf("--kubernetes (%s) does not specify a valid Kubernetes version. Supported versions: %s", k8sVersion, supportedVersions())
```

User should see their input in the error, shouldn't they?

Reply:
Makes sense. It _is_ confusing. My idea is that the error msg should show the user what the program "sees". The reason that the supported versions don't show a v prefix is because the user doesn't need to specify a v. 

Once we add the Semver type we should be able to loose all v prefixes the user sees.

Reply:
>My idea is that the error msg should show the user what the program "sees"

On the other hand, at the point where the error message is thrown the program sees the version prefixed with `v` (since `prefixedVersion` is actually the input to `K8sVersionFromMajorMinor`). And indeed this can be a bi*** from a development standpoint give that if you miss the `v` in the call to `K8sVersionFromMajorMinor`, you get back an empty string even if you enter a correct version.

So by the standpoint of what the program "sees" at this point, returning `prefixedVersion` would make the most sense. But yeah, that would mean adding `v` everywhere to `supportedVersions` (or rather: remove the TrimPrefix in this function).

I don't know what eventually the plan is to display versions across our tools. I think you know this better on what everyone agreed on, so you decide :)

Reply:
From what I have heard most want support for v-less versions. Internally we always need to add the v prefix. Lets go with the original input for now. I am happy to go back and change it if we come to a different conclusion later.